torch/storage.py

```diff
@@ -826,6 +859,13 @@ def hpu(self, device=None, non_blocking=False, **kwargs) -> T:  # type: ignore[m
         hpu_storage: torch.UntypedStorage = self._untyped_storage.hpu(device, non_blocking, **kwargs)
         return self._new_wrapped_storage(hpu_storage)
 
+    def to(self, device: torch.device, non_blocking: bool = False, **kwargs) -> T:  # type: ignore[type-var]
+        _warn_typed_storage_removal()
+        if self.dtype in [torch.quint8, torch.quint4x2, torch.quint2x4, torch.qint32, torch.qint8]:
+            raise RuntimeError("Cannot create XPU storage with quantized dtype")
```

Comment:
This needs removing I guess?
The c++ code should throw that error anyways right?

Reply:
I think it should, but whether underlying C++ cross will actually do that across 3 backends (cuda, hpu, xpu), I've no idea. Also, this might have been optimization to throw error earlier. I tend to leave this as is unless you will insist. (need to fix "XPU" in the message)

Reply:
Was this an existing error message? I can't find any reference to it.
And yes I do expect backends like cuda to definitely implement this properly.

Reply:
Actually, since that's a new API (`.to`), I think we can avoid having this thrown relying on c++ code. And if there will be a case when c++ does not throw, this can be considered an error. So, I will remove.

Reply:
> Was this an existing error message? I can't find any reference to it.

Sorry, I somehow missed this comment. Yes, it was existing message in `cuda()` and `hpu()` which we generalize into this new `to()`. See:
https://github.com/pytorch/pytorch/blob/3d20a5dd87e9c45e20dbc89210cfc63e7824b2b9/torch/storage.py#L850-L851

Reply:
Ho ok!
Then feel free to keep it with the updated generic device name!

Reply:
I left this change in `.to()`.